src/state/mint/hooks.ts

```diff
@@ -95,7 +95,7 @@ export function useDerivedMintInfo(
     [Field.CURRENCY_B]: independentField === Field.CURRENCY_A ? dependentAmount : independentAmount
   }
 
-  const token0Price = pair?.token0Price
+  const currentPrice = currencies[Field.CURRENCY_A]?.name === pair?.token0.name ? pair?.token0Price : pair?.token1Price
```

Comment:
comparing names is dangerous, use `currencyEquals`

Reply:
`currencyEquals` only accepts type `Currency` -- is there a utility for comparing a currency with a token from a Pair?

Reply:
...I figure this UI is already broken and the reliability of the comparison is pretty low-stakes, but let me know if there is a better way to do the comparison. Tests are passing :)

Reply:
`Token` extends `Currency`, you can use `currencyEquals` here.
Comparing name is *known* to be broken as well, so I would prefer to merge a proper fix because otherwise I will just have to push another commit to fix it.

Reply:
However you need to wrap the currency first before comparing